### ISSUE TITLE
Argument 'Type type' of ChopperClient.service function may be omitted.

### ISSUE THREAD

**Issue by 5turman** [lejard-h/chopper#22]
In function body we can use type parameter T instead of argument type to get corresponding service:
```dart
T service<T extends ChopperService>() {
    final s = _services[T];
    if (s == null) {
      throw Exception("Service of type '$type' not found.");
    }
    return s;
  }
```
It leads us to more concise code:
```dart
final api = chopperClient.service<UserApi>();
```
or
```dart
UserApi api = chopperClient.service();
```
Compare with old version:
```dart
final api = chopperClient.service<UserApi>(UserApi);
```

**Comment by lejard-h** [comment]
Yes it would be nice, it was not possible at the time I started chopper due to dart2js I think, it should be possible now

**Comment by MichaelDark** [comment]
@lejard-h Already fixed in 2.2.0. Time to close this issue
